tests/basic/autopilot_test.go

```diff
@@ -1885,6 +1887,193 @@ func scheduleAppsWithAutopilot(testName string, testScaleFactor int, apRules []a
+				wg.Add(1)
+
+				go func(nodeToCrash node.Node) {
+					defer wg.Done() // Decrement the WaitGroup counter when the task is done
+					stepLog := fmt.Sprintf("crash node: %s", nodeToCrash.Name)
+					Step(stepLog, func() {
+						log.InfoD(stepLog)
+						err := Inst().N.CrashNode(nodeToCrash, node.CrashNodeOpts{
+							Force: true,
+							ConnectionOpts: node.ConnectionOpts{
+								Timeout:         defaultCommandTimeout,
+								TimeBeforeRetry: defaultCommandRetry,
+							},
+						})
+						dash.VerifySafely(err, nil, "Validate node is crashed")
+					})
+
+					stepLog = fmt.Sprintf("wait for node: %s to be back up", nodeToCrash.Name)
+					Step(stepLog, func() {
+						log.InfoD(stepLog)
+						err := Inst().N.TestConnection(nodeToCrash, node.ConnectionOpts{
+							Timeout:         defaultTestConnectionTimeout,
+							TimeBeforeRetry: defaultWaitRebootRetry,
+						})
+						log.FailOnError(err, "Validate node is back up")
+					})
+
+					stepLog = fmt.Sprintf("wait for scheduler: %s and volume driver: %s to start",
+						Inst().S.String(), Inst().V.String())
+					Step(stepLog, func() {
+						log.InfoD(stepLog)
+						err := Inst().S.IsNodeReady(nodeToCrash)
+						log.FailOnError(err, "Validate node is ready")
+						err = Inst().V.WaitDriverUpOnNode(nodeToCrash, Inst().DriverStartTimeout)
+						log.FailOnError(err, "Validate volume driver is up")
+					})
+				}(nodeToCrash)
+			}
+			// Wait for all tasks to finish before proceeding
+			wg.Wait()
+
+			// calculate pool size from first node
+			provisionStatus, err := GetClusterProvisionStatusOnSpecificNode(crashedNodes[0])
+			log.FailOnError(err, "Failed to get cluster info for node:%v", crashedNodes[0])
+
+			var sizeAfterPoolExpand float64 = 0
+			for _, pstatus := range provisionStatus {
+				sizeAfterPoolExpand += pstatus.TotalSize
+			}
+			if sizeAfterPoolExpand <= poolSizes[0] {
+				err := errors.New("error pool expand failed")
+				log.FailOnError(err, "Pool expand failed")
+			}
+
+			//calculate pool size from second node
+			provisionStatus, err = GetClusterProvisionStatusOnSpecificNode(crashedNodes[1])
+			log.FailOnError(err, "Failed to get cluster info for node:%v", crashedNodes[1].Name)
+			log.InfoD("Pool expand successfully completed, size after pool expand:%v", sizeAfterPoolExpand)
+
+			sizeAfterPoolExpand = 0
+			for _, pstatus := range provisionStatus {
+				sizeAfterPoolExpand += pstatus.TotalSize
+			}
+
+			if sizeAfterPoolExpand <= poolSizes[1] {
+				err = errors.New("error pool expand failed")
+				log.FailOnError(err, "Pool expand failed")
+			}
+
+			log.InfoD("Pool expand successfully completed, size after pool expand:%v", sizeAfterPoolExpand)
+
+		})
+
+		Step("wait until workload completes on volume", func() {
+			log.InfoD("Volume workload started")
+			for _, ctx := range contexts {
+				err := Inst().S.WaitForRunning(ctx, workloadTimeout, retryInterval)
+				dash.VerifyFatal(err, nil, "Failed to wait for volume workload to run")
```

Comment:
ValidateContext?

Reply:
Instead of waitforrunning added validatecontext